### PR TITLE
Post team attendance to Slack weekly

### LINKED ISSUE #773: Post attendance without dues report in Slack automatically
Post to team private channels, weekly or so

### PR DESCRIPTION
**Pull request by JoshOldenburg**
Post a report weekly (Sundays at 11AM) to team private channels showing attendance in the past week and who hasn't paid dues. It only shows the people who haven't paid dues if there's an active dues package that's available for purchase.

Fixes #773.

image
image

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -56,3 +56,5 @@ JEDI_TOKEN=
 
 GITHUB_CLIENT_ID=
 GITHUB_CLIENT_SECRET=
+
+TEAM_SLACK_WEBHOOK_URL=
```
Comment (kberzinch) [inline .env.example]: Are you sure you can use a single Slack webhook for more than one channel?

File: app/Notifications/AttendanceNotification.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        $unknown = $team->attendance()->whereBetween('created_at', [$startDay, $endDay])->doesntHave('attendee')
+            ->selectRaw('count(distinct attendance.gtid) as aggregate')
+            ->get()[0]->aggregate;
+        $knownAttendance = $team->attendance()->whereBetween('created_at', [$startDay, $endDay])->has('attendee')
+            ->get();
+        $duesPackageAvailable = DuesPackage::availableForPurchase()->active()->count() > 0;
+
+        $inactiveNames = $knownAttendance->pluck('attendee')
+            ->unique()
+            ->filter(static function ($user): bool {
```
Comment (kberzinch) [inline app/Notifications/AttendanceNotification.php]: Type hint the parameter here

File: app/Notifications/AttendanceNotification.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        $inactiveNames = $inactiveNames->join(', ', ' and ');
+
+        $slackMessage = (new SlackMessage())
+            ->from(config('app.name'), ':robobuzz:')
+            ->to($team->slack_private_channel_id)
+            ->content($message);
+
+        if ($inactive > 0 && $duesPackageAvailable) {
+            $slackMessage = $slackMessage->warning()
+                ->attachment(static function ($attachment) use ($inactiveTitle, $inactiveNames): void {
```
Comment (kberzinch) [inline app/Notifications/AttendanceNotification.php]: Type hint the parameter here

File: .env.example (hunk 1 above)
Comment (JoshOldenburg) [reply .env.example]: Yes, the screenshot was made using the #onken-alerts hook but with the destination set to @joldenburg, so it came as a DM to me.

**Pull request merged**